src/interactivity/win32/windowio.cpp

```diff
@@ -167,6 +167,26 @@ void HandleKeyEvent(const HWND hWnd,
         // --- END LOAD BEARING CODE ---
     }
 
+    // - Certain applications like PowerToys and AutoHotKey, due to their keyboard
+    //   remapping feature, send us key events (WM_KEYUP, WM_KEYDOWN, etc.) using
+    //   SendInput(), whose keycode or scancode values could be outside of the
+    //   valid range. We need to filter those events out, as some clients will
+    //   incorrectly translate them to an ascii NULL character.
+    // - If scan code is 0, we can still infer it from the virtual key code as long
+    //   as the virtual key code is valid (0 < vKey < 0xFF).
+    if (VirtualScanCode == 0) [[unlikely]]
+    {
+        auto FullVirtualScanCode = gsl::narrow_cast<WORD>(OneCoreSafeMapVirtualKeyW(VirtualKeyCode, MAPVK_VK_TO_VSC_EX));
+        VirtualScanCode = LOBYTE(FullVirtualScanCode);
+        // If the virtual key code is invalid, translation fails with LOBYTE(0). (We ignore the enhanced bit because it doesn't affect the scan code.)
+        if (VirtualScanCode == 0)
+        {
+            return;
+        }
+        // Otherwise, set 'enhanced' bit if necessary and continue.
+        ControlKeyState |= (HIBYTE(FullVirtualScanCode) == 0xE0) ? ENHANCED_KEY : 0;
+    }

```

Comment:
Just one last question... Should we really run this code for `Message == WM_CHAR`?

Reply:
Nope. Just gave another try and found `VK_PACKET` based `SendInput` calls doesn't work if we apply this logic in case of `WM_CHAR`.

I'll update it to first check if it's not `WM_[SYS][DEAD]CHAR` before applying the filter logic.